azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/oauth/v2/filters/ClientCredentialsSecretFlow.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecomputearm.oauth.v2.filters;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Supplier;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.jclouds.azurecomputearm.oauth.v2.AzureAuthorizationApi;
+import org.jclouds.azurecomputearm.oauth.v2.domain.ClientSecret;
+import org.jclouds.oauth.v2.filters.OAuthFilter;
+import org.jclouds.oauth.v2.config.OAuthScopes;
+import org.jclouds.oauth.v2.domain.Token;
+import org.jclouds.domain.Credentials;
+import org.jclouds.http.HttpException;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.location.Provider;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.jclouds.Constants.PROPERTY_SESSION_INTERVAL;
+import static org.jclouds.azurecomputearm.oauth.v2.config.AzureOAuthProperties.RESOURCE;
+
+/**
+ * Authorizes new Bearer Tokens at runtime by sending up for the http request.
+ *
+ * <h3>Cache</h3>
+ * This maintains a time-based Bearer Token cache. By default expires after 59 minutes
+ * (the maximum time a token is valid is 60 minutes).
+ * This cache and expiry period is system-wide and does not attend to per-instance expiry time
+ * (e.g. "expires_in" from Google Compute -- which is set to the standard 3600 seconds).
+ */
+public class ClientCredentialsSecretFlow implements OAuthFilter {
+    private static final Joiner ON_COMMA = Joiner.on(",");
+
+    private final Supplier<Credentials> credentialsSupplier;
+    private final OAuthScopes scopes;
+    private final long tokenDuration;
+    private final String resource;
+    private final LoadingCache<ClientSecret, Token> tokenCache;
+
+    public static class TestClientCredentialsSecretFlow extends ClientCredentialsSecretFlow {
```

Comment:
Oh, then I'll take care of moving that class too to the test scope :) Thanks for spotting!

> But actually isn't needed as the tests cover functionality.

Then let's just remove it.